presto-product-tests/src/main/java/com/facebook/presto/tests/convention/TestShowPartitions.java

```diff
@@ -68,13 +84,98 @@ private static TableDefinition partitionedTableDefinition()
                 .build();
     }
 
+    private static TableDefinition partitionedTableWithVariablePartitionsDefinition()
+    {
+        String createTableDdl = "CREATE TABLE %NAME%(col INT) " +
+                "PARTITIONED BY (part_col INT) " +
+                "STORED AS ORC";
+
+        return HiveTableDefinition.builder(PARTITIONED_TABLE_WITH_VARIABLE_PARTITIONS)
+                .setCreateTableDDLTemplate(createTableDdl)
+                .setNoData()
+                .build();
+    }
+
     @Test(groups = {BASIC_SQL})
     public void testShowPartitionsFromHiveTable()
     {
         String tableNameInDatabase = tablesState.get(PARTITIONED_TABLE).getNameInDatabase();
 
-        String selectFromOnePartitionsSql = "show partitions from " + tableNameInDatabase;
-        QueryResult partitionListResult = query(selectFromOnePartitionsSql);
+        QueryResult partitionListResult;
+
+        partitionListResult = query("SHOW PARTITIONS FROM " + tableNameInDatabase);
         assertThat(partitionListResult).containsExactly(row(1), row(2));
+        assertColumnNames(partitionListResult, "part_col");
+
+        partitionListResult = query(format("SHOW PARTITIONS FROM %s WHERE part_col = 1", tableNameInDatabase));
+        assertThat(partitionListResult).containsExactly(row(1));
+        assertColumnNames(partitionListResult, "part_col");
+
+        assertThat(() -> query(format("SHOW PARTITIONS FROM %s WHERE no_such_column = 1", tableNameInDatabase)))
+                .failsWithMessage("Column 'no_such_column' cannot be resolved");
+        assertThat(() -> query(format("SHOW PARTITIONS FROM %s WHERE col = 1", tableNameInDatabase)))
+                .failsWithMessage("Column 'col' cannot be resolved");
```

Comment:
The error message is wrong. It should tell that `col` is not partitioned.

Reply:
Depends how you view it. If you say (as in docs) that `SHOW PARTITIONS` returns partition information that can be further narrowed down, then saying `'col' cannot be resolved` is really appropriate. 

btw, this is what used to be returned previously.